sdk/core/Azure.Core/src/Response.cs

```diff
@@ -142,13 +144,13 @@ public override string ToString()
             return $"Status: {Status}, ReasonPhrase: {ReasonPhrase}";
         }
 
-        internal static void DisposeStreamIfNotBuffered(ref Stream? stream)
+        internal void DisposeStreamIfNotBuffered(ref Stream? stream)
         {
             // We want to keep the ContentStream readable
             // even after the response is disposed but only if it's a
             // buffered memory stream otherwise we can leave a network
             // connection hanging open
-            if (stream is not MemoryStream)
+            if (stream is not MemoryStream && !ContentExtracted)
```

Comment:
We may rework this further in ClientModel, but I think this is satisfactory to unblock the current generator need.